Add Page model with encoding/decoding

pnwpedro opened the pull request:
<!-- Reminder: Keep READMEs up to date -->

## Problem

We were unwrapping `@set`, but not deserializing it into an object.

## Solution

* Add Page model
* Add decoding behavior
  * `{ @set: { data: ..., after: "..." }}` - materialized set
  * `{ @set: "..."` - unmaterialized set
* Raise on encode attempt because core does not yet support encoding to `@set`

This does not yet support nice pagination/iteration.

## Result

We can now decode a fauna set into a Page

## Testing

Integ and unit tests


----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

sprsquish commented:
> My test succeeds on core running locally. Perhaps we have to update the docker container.

Have you pulled recently? I ran into an issue with local tests only to realize I hadn't explicitly pulled a new version.

pnwpedro commented:
We updated the fauna container, but I started getting a 500, probably because my test query was recursive. I removed it for now, but kept the unit tests.

cynicaljoy commented on fauna/encoding/encoder.py:
```diff
@@ -1,7 +1,7 @@
 from typing import Any, Optional, Set
 
-from fauna.query.models import DocumentReference, Module, Document, NamedDocument, NamedDocumentReference
+from fauna.query.models import DocumentReference, Module, Document, NamedDocument, NamedDocumentReference, Page
```
`Page` is an unused import so that means it's missing from `_encode` ya?

pnwpedro replied:
oh yeah, i added an explicit error for encoding, hence the import, but then removed it to let it error like any other unencodable obj

nice catch
